src/Reflection/ReflectionFunctionAbstract.php

```diff
@@ -28,13 +26,107 @@
 
 trait ReflectionFunctionAbstract
 {
+    /** @var non-empty-string */
+    private string $name;
+
+    /** @var list<ReflectionParameter> */
+    private array $parameters;
+
+    private bool $returnsReference;
+
+    private ReflectionNamedType|ReflectionUnionType|ReflectionIntersectionType|null $returnType;
+
+    /** @var list<ReflectionAttribute> */
+    private array $attributes;
+
+    private string|null $docComment;
+
+    /** @var positive-int|null */
+    private int|null $startLine;
+
+    /** @var positive-int|null */
+    private int|null $endLine;
+
+    /** @var positive-int|null */
+    private int|null $startColumn;
+
+    /** @var positive-int|null */
+    private int|null $endColumn;
+
+    private bool $couldThrow = false;
+
+    private bool $isClosure   = false;
+    private bool $isGenerator = false;
+
     abstract public function __toString(): string;
 
+    /** @return non-empty-string */
     abstract public function getShortName(): string;
 
+    private function fillFromNode(MethodNode|Node\Stmt\Function_|Node\Expr\Closure|Node\Expr\ArrowFunction $node): void
+    {
+        $this->parameters       = $this->createParameters($node);
+        $this->returnsReference = $node->returnsByRef();
+        $this->returnType       = $this->createReturnType($node);
+        $this->attributes       = ReflectionAttributeHelper::createAttributes($this->reflector, $this, $node->attrGroups);
+        $this->docComment       = GetLastDocComment::forNode($node);
+        $this->couldThrow       = $this->computeCouldThrow($node);
+
+        if ($node->hasAttribute('startLine')) {
+            $startLine = $node->getStartLine();
+            assert($startLine > 0);
```

Comment:
Hmm, weird mutation: incompatible with type declaration.

I wonder if it's uncovered code not handled by pcov/phpdbg 🤔

Reply:
It's strange because the some code in other reflections is OK.